homeassistant/components/statistics/sensor.py

```diff
@@ -52,6 +52,7 @@
 DEFAULT_SIZE = 20
 DEFAULT_PRECISION = 2
 ICON = "mdi:calculator"
+UNIT_OF_MEASUREMENT_BINARY = "changes"
```

Comment:
`changes` is not an unit of measurement. The value is counting the state changes but that doesn't mean that we should start adding "what the sensors is doing" as unit of measurement to the values. If giving the sensor a proper name is not enough then one could further specify the unit of measurement (for a badge or a card) with "Customizations".

Reply:
I think there is a unit of measurement here but we might need to settle on a name. Consider that if you were to derive a rate such as "changes per hour" it would be immediately accurate and meaningful. For it to be unitless doesn't as accurately describe what's being measured.

My initial goal was to have it treated as the number it is rather than a discrete state string. In the process of the review noting that the unit is "changes" I felt added additional some clarity.

If the issue is the specific naming of the unit other meaningful units could be "counts", "state changes", "toggles", "transitions" all have similar meaning.

A search of scholarly papers for "changes per" yields abundant results which I think gives weight to the concept of "changes" as a unit. After some more searching I was also able to find references to SI-prefixed units such as "kilotransitions", "kilocounts", "kcounts", "gigacounts", "gcounts" in papers across various fields.

So if it were changed to use "counts" as a unit it would not only be meaningful it would also be consistent with the chosen unit name for similar measurements.

What do you say to using "counts" here?

Reply:
I realized I should add that when a `statistics` sensor is measuring a `sensor` type sensor that this doesn't occur because a `sensor` will always provide some unit of measurement even if that unit is `value` (which isn't really a unit, either, but it allows Home Assistant to interpret the type correctly and treat it like a number). So I think it's important to pick some name for this.

Reply:
I think I have another option that might apply here and I think it'll address the concern you have with specifying a unit by default. While also solving my issue that a `binary_sensor` won't have a `unit_of_measurement` for this sensor to adopt like it can for a normal `sensor`.

Using a configuration variable to explicitly specify a `unit_of_measurement` (similar to how one would configure a `template` sensor) will solve my problem of keeping a unit bound to my statistics sensor with minimal configuration pain while both avoiding changing the default behavior (making this no longer a breaking change) and avoid having to settle on a specific default unit.

While it is possible to override the missing attribute with customizations or wrapping it in a template sensor I think this is far more elegant and it makes configuration less error prone (it took me a few tries to get it to work as I expected when I was experimenting with it).

Reply:
We don't want to add config options for for unit of measurement. We should just make a decision.